torchao/dtypes/affine_quantized_tensor.py

```diff
@@ -194,30 +214,16 @@ def from_float(
         zero_point_dtype: Optional[torch.dtype] = None,
         preserve_zero: bool = True,
         zero_point_domain: ZeroPointDomain = ZeroPointDomain.INT,
-        extended_layout: str = "plain",
-        # TODO: this is only for "tensor_core_tiled", need to figure out
-        # the proper API for this arg
-        inner_k_tiles: Optional[int] = None,
+        layout_type: LayoutType = PlainLayoutType(),
     ):
         original_shape = input_float.shape
-        if extended_layout == "tensor_core_tiled":
-            orig_out_features, orig_in_features = input_float.shape
-            in_features = find_multiple(orig_in_features, 1024)
-            out_features = find_multiple(orig_out_features, 8)
-            input_float = torch.nn.functional.pad(
-                input_float,
-                (0, in_features - orig_in_features, 0, out_features - orig_out_features),
-            )
+        input_float = layout_type.pad_input(input_float)
```

Comment:
This is on the right track but I think we can make it more generic to better serve future usage. 
LayoutTypes can implement two functions: `pre_process()` and  `post_process()` The new workflow would look like this:
```
input_float = layout.pre_process(input_float)
...
int_data = quantize_affine(...)
int_data = layout.post_process(int_data)
layout_tensor_ctr = get_layout_tensor_constructor(type(layout_type))
layout_tensor = layout_tensor_ctr(int_data, scale, zero_point, layout_type)
```
To motivate this I can say that I would for sure use the post_process function while integrating my intx work. This allows me to write just a layout_type instead of an entire layout_class. Additionally, if you look at  #498 they call torch._cslt_compress(int_data) which could also be implemented as a post_process function.

Reply:
makes sense, by reusing PlainLayoutType, you mean inherit it and override `post_process` right?